components/modal/edit-expiry-modal.tsx

```diff
@@ -1,27 +1,57 @@
-import React, { useState } from 'react'
+import React, { useEffect, useMemo, useState } from 'react'
+import { useQuery } from 'wagmi'
 
 import Slider from '../slider'
-import { Currency } from '../../model/currency'
+import { LoanPosition } from '../../model/loan-position'
+import { fetchCouponAmountByEpochsBorrowed } from '../../api/market'
+import { useBorrowContext } from '../../contexts/borrow-context'
+import { useCurrencyContext } from '../../contexts/currency-context'
 
 import Modal from './modal'
 
-const dummy = [
-  { date: '24-06-30', profit: '102.37' },
-  { date: '24-12-31', profit: '102.37' },
-  { date: '25-06-30', profit: '102.37' },
-  { date: '25-12-31', profit: '102.37' },
-]
 const EditExpiryModal = ({
   position,
   onClose,
 }: {
-  position: {
-    currency: Currency
-    amount: string
-  } | null
+  position: LoanPosition
   onClose: () => void
 }) => {
+  const { balances } = useCurrencyContext()
+  const { extendLoanDuration, shortenLoanDuration } = useBorrowContext()
   const [selected, setSelected] = useState(0)
+
+  const { data } = useQuery(
+    ['coupon-amount-to-edit-expiry', position],
+    () => {
+      return fetchCouponAmountByEpochsBorrowed(
+        position.substitute,
+        position.amount,
+        position.toEpoch.id,
+      )
+    },
+    {
+      enabled: !!position,
```

Comment:
Isn't this always true?

Reply:
I've cherry picked this commit: https://github.com/clober-dex/coupon.finance/commit/8c99d59df7e34e10f88e316aadc72d4b99c2cd80